seqr/views/apis/summary_data_api_tests.py

```diff
@@ -78,33 +94,98 @@ def test_success_story(self, mock_analyst_group):
         response_json = response.json()
-        self.assertSetEqual(set(response_json.keys()), {
-            'projectsByGuid', 'locusListsByGuid', 'savedVariantsByGuid', 'variantFunctionalDataByGuid', 'genesById',
-            'variantNotesByGuid', 'individualsByGuid', 'variantTagsByGuid', 'familiesByGuid'})
+        self.assertSetEqual(set(response_json.keys()), SAVED_VARIANT_RESPONSE_KEYS)
         expected_variant_guids = {
-            'SV0000001_2103343353_r0390_100', 'SV0000007_prefix_19107_DEL_r00', 'SV0000006_1248367227_r0003_tes'}
+            'SV0000001_2103343353_r0390_100', 'SV0000007_prefix_19107_DEL_r00', 'SV0000006_1248367227_r0003_tes',
+        }
+        if self.MANAGER_VARIANT_GUID:
+            expected_variant_guids.add(self.MANAGER_VARIANT_GUID)
+        self.assertSetEqual(set(response_json['savedVariantsByGuid'].keys()), expected_variant_guids)
+
+        # Test analyst behavior
+        self.login_analyst_user()
+        response = self.client.get(gene_url)
+        self.assertEqual(response.status_code, 200)
+        response_json = response.json()
+        self.assertSetEqual(set(response_json.keys()), SAVED_VARIANT_RESPONSE_KEYS)
+        expected_variant_guids.discard(self.MANAGER_VARIANT_GUID)
         self.assertSetEqual(set(response_json['savedVariantsByGuid'].keys()), expected_variant_guids)
 
         all_tag_url = reverse(saved_variants_page, args=['ALL'])
         response = self.client.get('{}?gene=ENSG00000135953'.format(all_tag_url))
         self.assertEqual(response.status_code, 200)
         expected_variant_guids.add('SV0000002_1248367227_r0390_100')
         self.assertSetEqual(set(response.json()['savedVariantsByGuid'].keys()), expected_variant_guids)
+
+# Tests for AnVIL access disabled
+class LocalSummaryDataAPITest(AuthenticationTestCase, SummaryDataAPITest):
+    fixtures = ['users', '1kg_project', 'reference_data']
+    NUM_MANAGER_SUBMISSIONS = 4
+    MANAGER_VARIANT_GUID = 'SV0000006_1248367227_r0004_non'
+
+
+def assert_has_expected_calls(self, users, access_level_call=False):
+    calls = [mock.call(user) for user in users]
+    self.mock_list_workspaces.assert_has_calls(calls)
+    self.mock_get_ws_acl.assert_not_called()
+    if access_level_call:
+        self.mock_get_ws_access_level.assert_has_calls([
+            mock.call(self.manager_user, 'my-seqr-billing', 'anvil-project 1000 Genomes Demo'),
+            mock.call(self.manager_user, 'my-seqr-billing', 'anvil-1kg project n\u00e5me with uni\u00e7\u00f8de'),
+            mock.call(self.manager_user, 'my-seqr-billing', 'anvil-1kg project n\u00e5me with uni\u00e7\u00f8de'),
+            mock.call(self.manager_user, 'my-seqr-billing', 'anvil-1kg project n\u00e5me with uni\u00e7\u00f8de'),
+        ], any_order=True)
+    else:
+        self.mock_get_ws_access_level.assert_not_called()
+
+# Test for permissions from AnVIL only
+class AnvilSummaryDataAPITest(AnvilAuthenticationTestCase, SummaryDataAPITest):
```

Comment:
Do you think there still is a scenario that uses the AnVIL permissions only? If no, we could remove the AnVIL only tests. Maybe, we can remove them in a separate PR.

Reply:
yes, when we eventually take out the ability to log in locally it will be pure anvil and we will want this tested